googletest/include/gtest/internal/gtest-port.h

```diff
@@ -874,6 +874,23 @@ using ::std::tuple_size;
 # define GTEST_ATTRIBUTE_UNUSED_
 #endif
 
+// Use this annotation before a function that takes a printf format string.
```

Comment:
I can probably cut down the MinGW stuff if you don't care about that. I erred on the side of generality since it looked like you all supported a lot of things? MinGW has some weirdness due to it having two different `printf`s which depend on whether `__USE_MINGW_ANSI_STDIO` is set. It's a mess.